arangod/Aql/Query.cpp

```diff
@@ -2600,7 +2603,10 @@ void Query::toVelocyPack(velocypack::Builder& builder, bool isCurrent,
   builder.add("stream", VPackValue(queryOptions().stream));
 
   // modification query yes/no?
-  builder.add("modificationQuery", VPackValue(isModificationQuery()));
+  // if the query is cached there is no _ast
+  if (!_isCached) {
+    builder.add("modificationQuery", VPackValue(isModificationQuery()));
+  }
```

Comment:
I am also open to the idea of just returning an optional from `isModificationQuery` as well, but then we would have to handle it on all the places we perform a call. However, there are not a lot of places for that.